src/trees/mmt.rs

```diff
@@ -0,0 +1,812 @@
+//! Min-Max tree implementation as described by Cordova and Navarro in
+//! [Simple and efficient fully-functional succinct trees](https://doi.org/10.1016/j.tcs.2016.04.031).
```

Comment:
Here you mention the Cordova paper, but we don't actually implement it, do we? Is there a better paper to refer to?